web/pgadmin/tools/sqleditor/static/js/components/sections/ResultSet.jsx

```diff
@@ -559,12 +561,25 @@ export class ResultSetUtils {
     return columns;
   }
 
-  processClipboardVal(columnVal, col, rawCopiedVal) {
-    if(columnVal === '') {
-      if(col.has_default_val) {
-        columnVal = undefined;
-      } else if(rawCopiedVal === null) {
+  processClipboardVal(columnVal, col, rawCopiedVal, pasteSerials) {
+    if (col.isPK) {
```

Comment:
Do we really need to check PK? If a PK is having a serial type then it will automatically taken care with serial handling code. If it is not a serial then we should not do anything with copied value.

Reply:
We have two issues and in one of them user wants to clear the primary key when copy pasting that is the reason primary key check is there.

Reply:
As per the user `any table (with existing rows) that has an auto-generated PK (e.g., determined by a Sequence)`

Reply:
https://github.com/pgadmin-org/pgadmin4/issues/5922
If there is not default value or sequence being used then PK will be empty on paste.

Reply:
Why should the PK be empty if there no serial or sequence?
Also, both the tickets mention to clear only when there is a serial or auto generated val.